libr/bin/p/bin_symbols.c

```diff
@@ -289,13 +289,24 @@ static bool load_buffer(RBinFile *bf, void **bin_obj, RBuffer *buf, ut64 loadadd
 		eprintf ("Invalid headers\n");
 		return false;
 	}
-	printSymbolsHeader (sh);
+	//printSymbolsHeader (sh);
 	SymbolsMetadata sm = parseMetadata (buf, 0x40);
-	RCoreSymCacheElement *element = parseDragons (bf, buf, sm.addr + sm.size, sm.bits);
+	char * file_name = NULL;
+	if (sm.namelen) {
+		file_name = malloc (sm.namelen);
+		if (!file_name) {
+			return false;
+		}
+		if (r_buf_read_at (buf, 0x50, (ut8*)file_name, sm.namelen) != sm.namelen) {
```

Comment:
this can result on non-null terminated string, use calloc with namelen+1 for a simple fix

Reply:
good catch!

yeah in all samples i've seen the length actually includes the NUL terminator, but better be ready for it anyways.